components/Article/TableOfContents/index.tsx

```diff
@@ -0,0 +1,81 @@
+import { FormattedMessage } from 'react-intl';
+import Link from 'next/link';
+import styles from './index.module.scss';
+import type { FC } from 'react';
+
+type TableOfContentsNode = {
+  depth: number;
+  value: string;
+  id: string;
+  children?: TableOfContentsNode[];
+  parent?: TableOfContentsNode | null;
+};
+
+type TableOfContentsProps = {
+  tableOfContents: TableOfContentsNode[];
+};
+
+function parseTableOfContents(
```

Comment:
I see that your logic is a little bit based on how Nextra Headings work, but your logic is super over complicated and far from performance efficient. In simple words, it's utter mess.

I think there's a lot of room for improvement here (See the simplicity of Nextra's TOC component (https://github.com/shuding/nextra/blob/main/packages/nextra-theme-docs/src/components/toc.tsx)) the logic itself is just a simple `array.map` and nothing more.

Pretty much it uses a clever approach of using class names for different depths. And these class names adjust the left-marging and the font-size. I think that's honestly a really good and simple approach.

Reply:
We can do that but. We will not have nesting `<ul>`.
Just to know the pasing was done on front-end or in SSR ?

Reply:
We don't need `ul`.

> Just to know the pasing was done on front-end or in SSR ?

I'm not sure what you're saying.

Reply:
> I'm not sure what you're saying.

I mean it's client component or serveur component ?

Reply:
@AugustinMauroy 

Hi Augustin,

I just take a look at both codes (yours and Nextra's). There are many ways to implement/code things. 😅 

Something simpler is usually better (coding together is about teamwork). For example: I understand Nextra's code (and how it works) in under 2 mins (it is simply an `items.map()` and each item has `id`, `value`, and `depth`).

But, it takes more time to understand this PR. :)
What you write is called "Iterative DFS" (let's Not discuss about this, I just wanted to show that it is something with a cool name). And not all people know about it.

So, I think it's better to keep things simpler. What do you think?
Maybe you can try to look into how Nextra's implement the ToC and try to understand it first. :)
Try to understand why things work... It will be useful for you...

Reply:
I'm taking a look to css instead of this pice of incredible complexity

Reply:
Hey, @AugustinMauroy, any update based on @HinataKah0's feedback? 👀

Reply:
CSS part I have the solution. I'm looking on nextjs part. Because now we doesn't have nextra.

Reply:
Why would you need Nextra here? Their Navigation component is agnostic, I'm not seeing the issue here.

Reply:
The system was based on nextra toc.

Reply:
What system are you talking about? We only mentioned that our ToC can be a simple Array loop (1-dimensional) and the rest be CSS, like Nextra.

Reply:
Yeah but nextra give (in pagePros) an array with data but now we don't have this data so I have to create this data fetcher

Reply:
> Yeah but nextra give (in pagePros) an array with data but now we don't have this data so I have to create this data fetcher

I have no idea what you're talking about. The component should be agnostic to whatever. Yes we do not use Nextra, but we can easily get the ToC by using plugins, such as https://github.com/remarkjs/remark-toc.

In the end, the data structure we pass to the TOC component should be agnostic and can be whatever we define it to be. It should pretty much contain a `title`, `level` and `id`; This should be the data structure of any TOC that comes from Markdown.

So you shouldn't worry about what Framework we use or not, the TOC component should be simple, have a generic data structure, since we can pretty much when calling the component map any data structure from the Framework/plugin we use to what the component supports.